relay-dynamic-config/src/ai.rs

```diff
@@ -0,0 +1,78 @@
+//! Configuration for measurements generated from AI model instrumentation.
+
+use relay_common::glob2::LazyGlob;
+use serde::{Deserialize, Serialize};
+
+const MAX_SUPPORTED_VERSION: u16 = 1;
+
+#[derive(Clone, Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ModelCosts {
+    version: u16,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    costs: Vec<ModelCost>,
+}
+
+impl ModelCosts {
+    /// `false` if measurement and metrics extraction should be skipped.
+    pub fn is_enabled(&self) -> bool {
+        self.version > 0 && self.version <= MAX_SUPPORTED_VERSION
+    }
+
+    /// Gets the cost per 1000 tokens, if defined for the given model.
+    pub fn cost_per_1k_tokens(&self, model_id: &str, for_completion: bool) -> Option<f32> {
+        self.costs
+            .iter()
+            .find(|cost| cost.matches(model_id, for_completion))
+            .map(|c| c.cost_per_1k_tokens)
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ModelCost {
+    model_id: LazyGlob,
+    for_completion: bool,
```

Comment:
Maybe this makes sense as an enum, but not familiar enough with AI terminology:

```rs
enum Usage {
    Completion,
    NotCompletionButTheDefaultUsecase,
    // MaybeSomeFutureUsecase,
}
```

@colin-sentry wdyt?

Reply:
the SDKs only send the static `prompt_tokens`, `completion_tokens`, and `total_tokens`, so it's probably fine to keep as a boolean